### DISCUSSION TITLE
个人创建的路由必须要提交后才能在docker版rsshub中使用么？

### DISCUSSION THREAD

**Discussion by valuex** [DIYgod/RSSHub#16660] [Q&A] [answered]
个人使用的rsshub是部署在群晖nas的。在windows上调试好了路由，现在的问题是怎么把调试好的新路由更新到docker版的rsshub里。是不是必须要合入到rsshub 主程序中才能在docker中应用啊。
这个路由是个人使用的，另外typescript 也并不熟练，只是改了现存的route，能工作而已。所以并没打算合入到主程序里。有没有办法使得某些路由只能供个人使用（当然环境还得是docker 版rss，windows 本地调试通了但不方便tiny tiny rss 调用）？谢谢。

**Comment by valuex** [comment]
我把`namespace.ts`以及`handler.ts`文件拷贝到`rsshub`容器对应的目录下，重启容器后，加载对应的路由，但并没有成功。

![image](https://github.com/user-attachments/assets/bfa2a6b4-dd7e-4489-8a7a-51e632e388ce)

**Reply by wiketool** [reply, marked as answer]
可以在clone一份代码在本地使用Dockerfile进行构建；如果NAS的性能不足以支撑构建，使用其他设备构建好Docker image后可以考虑推送到阿某云/Dockerhub等仓库，再从仓库拉取镜像

**Reply by valuex** [reply]
感谢回复，回头试试。  
Javascript或Typescript是不需要编译的。现在要重新构建一下，再安装到NAS中。从动作上看就类似于编译再发布了。  
而在本地版本中（在nodejs中运行的版本）中，可在创建路由后，重新启动下`npm dev` 就可以可以运行了。  
不知道在docker版本中，能不能实现类似的效果，即将用户路由放到docker容器挂载的外部文件夹中，重启容器就可看到效果了。

**Reply by bestlzk** [reply]
可以试下在容器内执行`npm run build`

**Comment by sunkunkun** [comment]
我将用户路由放到docker容器挂载的外部文件夹中，重启容器不行，请问你后面是如何实现使用个人创建的路由的？

**Reply by synesios** [reply]
我也是，你解决了吗

**Comment by eve2ptp** [comment]
参考 Dockerfile 挂载源码目录，在 node 容器跑 pnpm dev
